local_pathfinding/node_mock_global_path.py

```diff
@@ -0,0 +1,449 @@
+            write (bool, optional):
+                Whether to write the path to a new csv file, default False
+            file_path (str, optional):
+                The filepath to the csv file containing the global path,
+
+        Returns:
+            Path: The interpolated path
+        """
+
+        waypoints = [pos] + global_path.waypoints
+
+        i, j = 0, 0
+        while i < len(path_spacing):
+            if path_spacing[i] > interval_spacing:
+                # interpolate a new sub path between the two waypoints
+                pos = waypoints[j]
+                dest = waypoints[j + 1]
+
+                sub_path = MockGlobalPath.generate_path(
+                    dest=dest,
+                    interval_spacing=interval_spacing,
+                    pos=pos,
+                )
+                # insert sub path into path
+                waypoints[j + 1 : j + 1] = sub_path.waypoints[:-1]
+                # shift indices to account for path insertion
+                j += len(sub_path.waypoints) - 1
+
+            i += 1
+            j += 1
+        # remove pos from waypoints again
+        waypoints.pop(0)
+
+        global_path.waypoints = waypoints
+
+        if write:
+            MockGlobalPath.write_to_file(file_path=file_path, global_path=global_path)
+
+        return global_path
+
+    @staticmethod
+    def interval_spacing(pos: HelperLatLon, waypoints: list[HelperLatLon]) -> list[float]:
+        """Returns the distances between pairs of points in a list of latitudes and longitudes,
+        including pos as the first point.
+
+        Args:
+            pos (HelperLatLon):
+                The gps position of the boat
+            waypoints (list[HelperLatLon]):
+                The list of waypoints
+
+        Returns:
+            list[float]: The distances between pairs of points in waypoints [km]
+        """
+        all_coords = [(pos.latitude, pos.longitude)] + [
+            (waypoint.latitude, waypoint.longitude) for waypoint in waypoints
+        ]
+
+        coords_array = np.array(all_coords)
+
+        lats1, lons1 = coords_array[:-1].T
+        lats2, lons2 = coords_array[1:].T
+
+        distances = GEODESIC.inv(lats1=lats1, lons1=lons1, lats2=lats2, lons2=lons2)[2]
+
+        distances = [meters_to_km(distance) for distance in distances]
+
+        return distances
+
+    @staticmethod
+    def write_to_file(file_path: str, global_path: Path, tmstmp=True) -> Path:
+        """Writes the global path to a new, timestamped csv file.
+
+        Args:
+            file_path (str):
+                The filepath to the csv file containing the global path,
+            global_path (Path):
+                The global path to write to file
+        """
+        if file_path == "":
+            raise ValueError("file_path must be specified when write is True")
+
+        # check if file_path is a valid file path
+        if "global_paths" not in file_path or not os.path.isdir(os.path.dirname(file_path)):
```

Comment:
If we want it to be in global_paths should we check if the directory name ends in `global_paths`?

Reply:
For sure. Also, do you think its neccessary to check isdir as well?

Reply:
Yes that is good, can only write to a directory that exists